src/main/java/de/sakpaas/backend/service/LocationService.java

```diff
@@ -273,4 +283,34 @@ private void createNewLocation(OsmResultLocationListDto.OsmResultLocationDto osm
     );
     this.save(location);
   }
+
+  /**
+   * Searches in the Nominatim Microservice for the given key.
+   *
+   * @param key The search parameter. Multiple words are separated with %20.
+   * @return The list of Locations in our database
+   */
+  public List<Location> search(String key) {
+    // Makes a request to the Nominatim Microservice
+    final String url = this.searchApiUrl + "/search/" + key + "?format=json";
```

Comment:
Das sehe ich als sehr unsicher an, wenn man das Argument einfach dazwischen klebt. Gibt es in Spring eine bessere Variante @benfu99 oder @Fabsi23?